mongodbatlas/resource_mongodbatlas_advanced_cluster.go

```diff
@@ -1124,6 +1125,36 @@ func resourceClusterAdvancedRefreshFunc(ctx context.Context, name, projectID str
 	}
 }
 
+func resourceClusterListAdvancedRefreshFunc(ctx context.Context, projectID string, client *matlas.Client) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		clusters, resp, err := client.AdvancedClusters.List(ctx, projectID, nil)
+
+		if err != nil && strings.Contains(err.Error(), "reset by peer") {
```

Comment:
[q] Can you use `errors.Is` here?

Reply:
I'm not sure if I can use `errors.Is` out of the box, is `err.Error() == "reset by peer"`? Or does it just contain it? 

Even if it's just containment, I could theoretically implement something like the following, would that be preferable?
```
type ErrorContains {
    Message string
}

func (e *ErrorContains) Error() string {
    return e.Message
}

func (e *ErrorContains) Is(tgt error) bool {
    return strings.Contains(tgt.Error(), e.Message)
}

var ResetByPeerError = &ErrorContains{ Message = "reset by peer" }
```

Reply:
I'm resolving this for now, but am open to improving this down the road depending on your response.